week-1-ts-ecosystem/w01e03-object-oriented-structure/src/basket/basket.ts

```diff
@@ -0,0 +1,50 @@
+import {Product} from "../product/product";
+import {ProductId} from "../product/models/product-properties.model";
+
+export class Basket<Type extends Product> {
```

Comment:
Tutaj drobny detal. Zmieniłbym `Type` na `TYPE` pisane z dużych (to taka konwencja dla generyków, z którą na pewno już sie spotkałeś i jeszcze spotkasz)

a ponieważ `TYPE` jest mocno ogólne może być nawet `PRODUCT`